awscrt/s3.py

```diff
@@ -274,17 +267,10 @@ def __init__(
 
         super().__init__()
 
-        self._on_headers_cb = on_headers
```

Comment:
Were these just unused?

Reply:
`S3Request` don't need to hold a reference to them anymore, since the `_S3RequestCore` handles everything😀